pkg/errors/errors_test.go

```diff
@@ -23,3 +38,2006 @@ func TestErrorFind(t *testing.T) {
+				err: &ServiceError{},
+			},
+			want: compat.Error{
+				Id:          strconv.Itoa(int(e.Code)),
+				Kind:        "Error",
+				Href:        Href(e.Code),
+				Code:        CodeStr(e.Code),
+				Reason:      e.Reason,
+				OperationId: "",
+			},
+		},
+		{
+			name: "should return compat.Error",
+			fields: fields{
+				err: &ServiceError{},
+			},
+			args: args{
+				basePath: "/api/connector_mgmt/",
+			},
+			want: compat.Error{
+				Id:          strconv.Itoa(int(e.Code)),
+				Kind:        "Error",
+				Href:        strings.Replace(Href(e.Code), ERROR_HREF, CONNECTOR_MGMT_ERROR_HREF, 1),
+				Code:        strings.Replace(CodeStr(e.Code), ERROR_CODE_PREFIX, CONNECTOR_MGMT_ERROR_CODE_PREFIX, 1),
+				Reason:      e.Reason,
+				OperationId: "",
+			},
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			Expect(tt.fields.err.AsOpenapiError(tt.args.operationID, tt.args.basePath)).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_StackTrace(t *testing.T) {
+	type fields struct {
+		err *ServiceError
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   errors.StackTrace
+	}{
+		{
+			name: "should return error stacktrace if error cause is nil",
+			fields: fields{
+				err: &ServiceError{},
+			},
+			want: nil,
+		},
+		{
+			name: "should return error stacktrace if cause defined",
+			fields: fields{
+				err: &ServiceError{
+					cause: errorWithCause.cause,
+				},
+			},
+			want: errorWithCause.StackTrace(),
+		},
+		{
+			name: "should return nil if the cause doesn't have stacktrace",
+			fields: fields{
+				err: &ServiceError{
+					cause: &errorWithoutStackTrace{},
+				},
+			},
+			want: nil,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			Expect(tt.fields.err.StackTrace()).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_AsError(t *testing.T) {
+	type fields struct {
+		err *ServiceError
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   error
+	}{
+		{
+			name: "should return a formatted error",
+			fields: fields{
+				err: &ServiceError{
+					Code:     ServiceErrorCode(e.HttpCode),
+					Reason:   e.Reason,
+					HttpCode: e.HttpCode,
+					cause:    e.cause,
+				},
+			},
+			want: fmt.Errorf(e.Error()),
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			Expect(tt.fields.err.AsError()).To(Equal(tt.want))
+		})
+	}
+}
+
+func Test_Error(t *testing.T) {
+	errList := make(ErrorList, 10)
+	var res string
+	for _, err := range errList {
+		res = res + fmt.Sprintf(";%s", err)
+	}
+	type fields struct {
+		err ErrorList
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   string
+	}{
+		{
+			name: "should return a formatted error list",
+			fields: fields{
+				err: make(ErrorList, 10),
```

Comment:
I think `errList` created at the beginning of the test could be used here instead of creating new error list